[log4cxx] INTERFACE_LINK_LIBRARIES tries to link against EXPAT::EXPAT erroneously.

**Describe the bug**
The log4cxx package cannot be linked against on macOS with CMake, the generation failing to link against the target 'EXPAT::EXPAT'. (An XML library that log4cxx depends on.) The correct target name is 'expat::expat' (lower case).

This can be worked around by replacing EXPAT::EXPAT in the installed log4cxxConfig.cmake with expat::expat.

**Environment**
- OS: macOS (M2) Sonoma 14.1
- Compiler:
Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: arm64-apple-darwin23.1.0
- CMake: 3.28.0-rc3

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install log4cxx
2. Create a CMake project with an empty main.cpp file and the following CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.21)
project(Hello)

set(CMAKE_CXX_STANDARD 20)

find_package(log4cxx CONFIG REQUIRED)

add_executable(Hello main.cpp)

target_link_libraries(Hello PRIVATE log4cxx)
```
3. Then in a build directory, `cmake -DVKPKG_MANIFEST_MODE=OFF -DCMAKE_TOOLCHAIN_FILE=/Path/To/Vcpkg/scripts/buildsystems/vcpkg.cmake ..`

**Expected behavior**
CMake generation should succeed.

**Failure logs**
```
CMake Error at /Users/bernard/Source/vcpkg2/installed/arm64-osx/share/log4cxx/log4cxxConfig.cmake:62 (set_target_properties):
  The link interface of target "log4cxx" contains:

    EXPAT::EXPAT

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /Users/bernard/Source/vcpkg2/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  CMakeLists.txt:6 (find_package)
```

**Additional context**
The package works on Linux, at least.

--- a/ports/log4cxx/fix-find-package.patch
+++ b/ports/log4cxx/fix-find-package.patch
@@ -1,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 61c0479..3bd3327 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -60,7 +60,7 @@ find_package(APR-Util REQUIRED)
+ find_package( Threads REQUIRED )
+ 
+ # Find expat for XML parsing
+-find_package(EXPAT REQUIRED)
++find_package(expat CONFIG REQUIRED)
+ if(TARGET EXPAT::EXPAT)
+   set(EXPAT_LIBRARIES EXPAT::EXPAT)
+ elseif(TARGET expat::expat)

--- a/ports/log4cxx/portfile.cmake
+++ b/ports/log4cxx/portfile.cmake
@@ -6,6 +6,8 @@ vcpkg_download_distfile(ARCHIVE
 
 vcpkg_extract_source_archive(
     SOURCE_PATH ARCHIVE "${ARCHIVE}"
+    PATCHES
+        fix-find-package.patch
 )
 
 vcpkg_cmake_configure(

--- a/ports/log4cxx/vcpkg.json
+++ b/ports/log4cxx/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "log4cxx",
   "version": "1.1.0",
+  "port-version": 1,
   "description": "Apache log4cxx is a logging framework for C++ patterned after Apache log4j, which uses Apache Portable Runtime for most platform-specific code and should be usable on any platform supported by APR",
   "homepage": "https://logging.apache.org/log4cxx",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5214,7 +5214,7 @@
     },
     "log4cxx": {
       "baseline": "1.1.0",
-      "port-version": 0
+      "port-version": 1
     },
     "loguru": {
       "baseline": "2.1.0",

--- a/versions/l-/log4cxx.json
+++ b/versions/l-/log4cxx.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c030a796829f4525b2369a135d7bdc615b6fb14b",
+      "version": "1.1.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "eca29ecb5127d26bd46aad143468b069a045d104",
       "version": "1.1.0",
       "port-version": 0

